lib/src/rules/omit_local_variable_types.dart

```diff
@@ -79,7 +79,7 @@ class _Visitor extends SimpleAstVisitor<void> {
               DartTypeUtilities.isInterface(type, 'Iterable', 'dart.core'));
       if (iterableInterfaces.length == 1 &&
           iterableInterfaces.first.typeArguments.first == staticType.type) {
-        rule.reportLint(node);
+        rule.reportLint(node.loopVariable.type ?? node);
```

Comment:
This looks overly defensive.
We have already tested above at the line `71` that `staticType = node.loopVariable?.type` is not `null`. So, `node.loopVariable` also cannot be `null`.

Reply:
Good catch.

Reply:
And better still, we have the value cached already in `staticType`! 🎉